46.md

```diff
@@ -0,0 +1,162 @@
+NIP-46
+======
+
+Nostr Connect
+------------------------
+
+`draft` `optional` `author:tiero` `author:giowe` `author:vforvalerio87`
+
+## Rationale
+
+Private keys should be exposed to as few systems - apps, operating systems, devices - as possible as each system adds to the attack surface.
+
+Entering private keys can also be annoying and requires exposing them to even more systems such as the operating system's clipboard that might be monitored by malicious apps.
+
+
+## Terms
+
+* **App**: Nostr app on any platform that *requires* to act on behalf of a nostr account.
+* **Signer**: Nostr app that holds the private key of a nostr account and *can sign* on its behalf.
+
+
+## `TL;DR`
+
+
+**App** and **Signer** sends ephemeral encrypted messages to each other using kind `24133`, using a relay of choice. 
+
+App prompts the Signer to do things such as fetching the public key or signing events.
+
+The `content` field must be an encrypted JSONRPC-ish **request** or **response**.
+
+## Signer Protocol
+
+### Messages
+
+#### Request
+
+```json
+{
+  "id": <random_string>,
+  "method": <one_of_the_methods>,
+  "params": [<anything>, <else>]
+}
+```
+
+#### Response
+
+```json
+{
+  "id": <request_id>,
+  "result": <anything>,
+  "error": <reason>
+}
+```
+
+### Methods
+
+
+#### Mandatory 
+
+These are mandatory methods the remote signer app MUST implement:
+
+- **describe**
+  - params []
+  - result `{"get_public_key": { params: [], result: anything }}`  
+- **get_public_key**
```

Comment:
140 comments and I don't see anyone mentioning this uses underscores instead of camel case. :sob: I love this nip but it's going to drive me slightly crazy the function names don't match the equivalent NIP-07

Reply:
Probably because of `created_at` field from NIP-01

Reply:
PRs are welcome! I would be happy to amend it.